src/JellyfishClient.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  Metadata,
+  ConnectionOptions,
+} from '@jellyfish-dev/react-native-membrane-webrtc';
+import { useEffect, useRef } from 'react';
+import { NativeModules, NativeEventEmitter, Platform } from 'react-native';
+
+const LINKING_ERROR =
+  `The package 'react-native-membrane' doesn't seem to be linked. Make sure: \n\n` +
+  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo managed workflow\n';
+
+const Membrane = NativeModules.Membrane
+  ? NativeModules.Membrane
+  : new Proxy(
+      {},
+      {
+        get() {
+          throw new Error(LINKING_ERROR);
+        },
+      }
+    );
+
+const eventEmitter = new NativeEventEmitter(Membrane);
+
+/**
+ * A hook used to interact with jellyfish server.
+ */
+export function useJellyfishClient() {
+  const websocket = useRef<WebSocket | null>(null);
+
+  const sendMediaEvent = (mediaEvent: string) => {
+    if (websocket.current?.readyState !== WebSocket.OPEN) {
+      return;
+    }
+    const messageJS = {
+      type: 'mediaEvent',
+      data: mediaEvent,
+    };
+    websocket.current?.send(JSON.stringify(messageJS));
+  };
+
+  useEffect(() => {
+    const eventListener = eventEmitter.addListener(
+      'SendMediaEvent',
+      sendMediaEvent
+    );
+    return () => eventListener.remove();
+  }, []);
+
+  /**
+   * Connects to the server using the websocket connection.
+   *
+   * @param url - websocket url
+   * @param peerToken - token used to authenticate when joining the room
+   * @param connectionOptions - Configuration object for the connection
+   */
+  const connect = async (
+    url: string,
+    peerToken: string,
+    connectionOptions: Partial<ConnectionOptions> = {}
+  ) => {
+    websocket.current = new WebSocket(url);
+
+    websocket.current.addEventListener('open', () => {
+      console.log('open');
+    });
+    websocket.current.addEventListener('error', (err) => {
+      console.log('error', err);
```

Comment:
User should be able to get this error. For example when there is no internet the socket can't connect and it should return error here.
In the ts library we have this: https://github.com/jellyfish-dev/ts-client-sdk/blob/main/src/JellyfishClient.ts#L31
in this one we can return it in the hook (webrtc library does something similar, right?)

Reply:
Added an error state